### PR TITLE
[inductor] bug fix: use `create_symbolic_sizes_strides_storage_offset`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #94031

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/graph.py [hunk 1]
```diff
@@ -72,7 +72,17 @@ def symbolic_sizes_strides(self, ex: torch.Tensor):
...
+
+            # TODO: make a dedicated UnknownSource for this?
+            source = ConstantSource(
+                f"__unknown_tensor_{len(self._shape_env.var_to_val)}"
+            )
+            (
+                size,
+                stride,
+                _,
+            ) = self._shape_env.create_symbolic_sizes_strides_storage_offset(ex, source)
```
Comment (nkaretnikov) [inline torch/_inductor/graph.py:85]: `create_symbolic_sizes_strides` no longer exists, was replaced by this.

What's the proper way to create a source here? I've pretty much copy-pasted from `create_symint` in `/test/test_dynamic_shapes.py`.

File: test/inductor/test_torchinductor.py [hunk 2]
```diff
@@ -5447,7 +5447,6 @@ def fn(x):
     "test_cudnn_rnn_dynamic_shapes": ("cuda",),
     "test_grid_sampler_2d_dynamic_shapes": ("cpu", "cuda"),
     "test_kwargs_dynamic_shapes": ("cpu",),
-    "test_list_clearing_dynamic_shapes": ("cpu", "cuda"),
```
Comment (nkaretnikov) [inline test/inductor/test_torchinductor.py:5450]: Fixes these tests as well.

File: torch/_inductor/graph.py (hunk 1 above)
Comment (ezyang) [reply torch/_inductor/graph.py:85]: @Chillee and I discussed this and this code should become unreachable when https://github.com/pytorch/pytorch/pull/93059 lands. So I'd also look positively on you turning this into an assert False, or the hack is fine as well and just need to TODO it to be removed with @Chillee's patch.

**Review by ezyang — APPROVED** [review]
this will get overwritten shortly but whatever. @Chillee feel free to NACK this patch

**Pull request closed without merging**